### PR TITLE
VLAN enhancement

### PR DESCRIPTION
**Pull request by yaocw2020**
Related issue: https://github.com/harvester/harvester/issues/2236
Dependency PR: https://github.com/harvester/network-controller-harvester/pull/59

### CODE REVIEW

File: charts/harvester-network-controller/templates/rbac.yaml [hunk 1]
```diff
@@ -82,3 +85,59 @@ subjects:
...
+  {{- include "harvester-network-controller.labels" . | nindent 4 }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: harvester-network-webhook
+rules:
+  - apiGroups: [ "" ]
+    resources: [ "secrets", "configmaps" ]
+    verbs: [ "*" ]
```
Comment (guangbochen) [inline charts/harvester-network-controller/templates/rbac.yaml:104]: may consider merging webhook RBAC with the network as one, or u should remove those unneeded permissions.

File: charts/harvester-network-controller/values.yaml [hunk 2]
```diff
@@ -60,3 +60,18 @@ helper:
   image:
     repository: rancher/harvester-network-helper
     tag: "master-head"
+
+webhook:
+  replicas: 2
```
Comment (guangbochen) [inline charts/harvester-network-controller/values.yaml:65]: probably 1 is enough as don't want to increase the default system resource too much in the newer versions.

**Comment by guangbochen** [comment]
Also please fix the lint errors.

File: charts/harvester-network-controller/templates/rbac.yaml (hunk 1 above)
Comment (yaocw2020) [reply charts/harvester-network-controller/templates/rbac.yaml:104]: They are all needed.

File: charts/harvester-network-controller/templates/deployment.yaml [hunk 3]
```diff
@@ -64,7 +64,46 @@ spec:
...
+      {{- include "harvester-network-webhook.selectorLabels" . | nindent 6 }}
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        {{- include "harvester-network-webhook.selectorLabels" . | nindent 8 }}
+    spec:
+      serviceAccountName: harvester-network-webhook
+      hostNetwork: true
```
[the remark this answers is not in the document]
Comment (yaocw2020) [reply charts/harvester-network-controller/templates/deployment.yaml:96]: Good catch. I forget modifying it after my testing.

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**